### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by mishun**
Well, .NET 7 is out. I've found out that locally built tests under it are freezing or throw weird exception from insides of protobuf-net.

Not using default behavior to register Optional<_> seems to help with exception (I'm not sure what was happening there; maybe it's protobuf-net bug, maybe something was wrong with dependencies),

As for tests, I added .net7.0 to test workflow, updated dependencies, added FsCheck/Expecto integration and integration with dotnet's standard test framework (so tests could be run with `dotnet test`) and removed duplicated code.

### CODE REVIEW

File: src/ProtoBuf.FSharp/Surrogates.fs [hunk 1]
```diff
@@ -21,5 +21,10 @@ type Optional<'t> =
         | true -> Some w.Item
 
     static member RegisterIntoModel (model : RuntimeTypeModel) =
-        let surrogateModelType = model.Add(typeof<Optional<'t>>, true)
+        // For some reason on .net 7 that throws:
+        // System.TypeLoadException: Could not load type 'System.Runtime.CompilerServices.IsReadOnlyAttribute' from assembly 'System.Collections.Immutable, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
+        //let surrogateModelType = model.Add(typeof<Optional<'t>>, true)
```
Comment (mvkara) [inline src/ProtoBuf.FSharp/Surrogates.fs:26]: Probably some reflection magic or rather.

File: paket.dependencies [hunk 2]
```diff
@@ -1,7 +1,10 @@
...
+storage:none
+source https://www.nuget.org/api/v2
+
+nuget FSharp.Core >= 4 redirects: force
+nuget protobuf-net >= 3.0
+nuget FsCheck >= 2.0 < 3.0
+nuget Expecto >= 9.0 < 10.0
+nuget Expecto.FsCheck >= 9.0 < 10.0
+nuget YoloDev.Expecto.TestSdk >= 0.13 < 0.14
+nuget Microsoft.NET.Test.Sdk
```
Comment (mvkara) [inline paket.dependencies:10]: You shouldn't need to bring this in via Paket - it usually gets imported via the ProjectType in the *.fsproj file I thought.

**Review by mvkara — APPROVED** [review]
LGTM

**Comment by mvkara** [comment]
Thanks for the contribution, will merge the PR.

File: paket.dependencies (hunk 2 above)
Comment (mishun) [reply paket.dependencies:10]: If you mean `Microsoft.NET.Test.Sdk` part to make `dotnet test` work specifically, it was different a few years ago (if I remember correctly), but seems to have stabilized at importing `Microsoft.NET.Test.Sdk` as any other nuget package now --- at least it works from netcore3.1 to net7.0

File: src/ProtoBuf.FSharp/Surrogates.fs (hunk 1 above)
Comment (mishun) [reply src/ProtoBuf.FSharp/Surrogates.fs:26]: Yeah, it really bizarre --- I looked up place inside protobuf-net that was throwing and it seems to have nothing to do with System.Collections.Immutable (unless net7.0 uses it internally)

**Pull request merged**